src/main/java/budgetbuddy/ui/MainWindow.java

```diff
@@ -156,11 +155,23 @@ void fillInnerParts() {
     /**
      * Updates the current tab on display
      */
-    private void updateView(PanelTab tabToView) {
-        if (tabToView != null) {
-            TabPane pane = (TabPane) outputDisplayPlaceholder.getChildren().get(0);
-            pane.getSelectionModel().select(tabToView);
+    private void updateView(CommandCategory category) {
+        requireNonNull(category);
+
+        PanelTab tabToView = tabMap.get(
+                category.isSecondaryCategory()
+                        ? category.getPrimaryCategory(category)
+                        : category);
+
+        requireNonNull(tabToView);
```

Comment:
I don't think you should requireNonNull here. Commands like `help` or `exit` will have exceptions thrown in that case. Those commands should have categories but no tabs. Only update the current tab if the tabToView isn't null

Reply:
That’s a good point, I’ll move the checking of category back down and change the parameter here back to the tab to view.